### PR TITLE
fix #3525 writing to sys.stdout with --json flag for post-link scripts

### LINKED ISSUE #3525: Invalid JSON output
When installing `Jupyter` I sometimes see the following error:

```
post-link :: /etc/machine-id not found ..
bus post-link :: .. using /proc/sys/kernel/random/boot_id
```

When installing with the `--json` flag the error output causes the json to be invalid.  Example:

```
root@head:~# conda create -n test_env2 python jupyter -y --json -q
dbus post-link :: /etc/machine-id not found ..
dbus post-link :: .. using /proc/sys/kernel/random/boot_id
{
  "actions": {
    "LINK": [
      "expat-2.1.0-0 1",
      ...
    ],
    "PREFIX": "/opt/a/b/c/muunitnoc/anaconda/envs/test_env2",
    "SYMLINK_CONDA": [
      "/opt/a/b/c/muunitnoc/anaconda"
    ],
    "op_order": [
      "RM_FETCHED",
      "FETCH",
      "RM_EXTRACTED",
      "EXTRACT",
      "UNLINK",
      "LINK",
      "SYMLINK_CONDA"
    ]
  },
  "success": true
}
```

In my opinion this is a fairly critical -- I need to depend on valid JSON output

cc @kalefranz @koverholt @mingwandroid

### PR DESCRIPTION
**Pull request by kalefranz**
resolves #3525 

@quasiben Can you please review?

### CODE REVIEW

**Comment by kalefranz** [comment]
CC @ilanschnell

**Comment by kalefranz** [comment]
@quasiben You can thank (blame?) @goanpeca for getting this out and included in the 4.2 anaconda distribution release :-p

It'll be in conda 4.2.9 released in an hour or two.

File: conda/install.py [hunk 1]
```diff
@@ -1046,7 +1046,8 @@ def messages(prefix):
     path = join(prefix, '.messages.txt')
     try:
         with open(path) as fi:
-            sys.stdout.write(fi.read())
+            fh = sys.stderr if context.json else sys.stdout
+            fh.write(fi.read())
```
Comment (kalefranz) [inline conda/install.py]: If this weren't needed as a quick patch, I'd actually transition it to using loggers properly.  `TODO` though.

**Comment by quasiben** [comment]
Small local tests are working.  Going to move onto larger integration test.

Thanks so much @kalefranz @goanpeca for the fast response times!

**Comment by goanpeca** [comment]
What did I do :-p ?

**Pull request merged**